pkg/kubectl/stop.go

```diff
@@ -270,10 +271,70 @@ func (reaper *JobReaper) Stop(namespace, name string, timeout time.Duration, gra
 		return utilerrors.NewAggregate(errList)
 	}
 	// once we have all the pods removed we can safely remove the job itself
-	if err := jobs.Delete(name, gracePeriod); err != nil {
+	return jobs.Delete(name, gracePeriod)
+}
+
+func (reaper *DeploymentReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *api.DeleteOptions) error {
+	deployments := reaper.Extensions().Deployments(namespace)
+	replicationControllers := reaper.ReplicationControllers(namespace)
+
+	deployment, err := deployments.Get(name)
+	if err != nil {
+		return err
+	}
+
+	// set deployment's history and scale to 0
+	zero := 0
+	deployment.Spec.RevisionHistoryLimit = &zero
+	deployment.Spec.Replicas = 0
+	deployment, err = deployments.Update(deployment)
```

Comment:
I don't have Patch operation available at this point, since I'm interacting with client.Interface at this level.